crates/block-producer/src/block_builder/mod.rs

```diff
@@ -114,7 +114,9 @@ where
             .await?;
 
         let missing_notes: Vec<_> = dangling_notes
-            .difference(&block_inputs.found_unauthenticated_notes)
+            .difference(
+                &block_inputs.found_unauthenticated_notes.note_proofs.keys().copied().collect(),
```

Comment:
I would probably add a method on `NoteAuthenticationInfo` to make this a single method invocation (e.g., maybe something like `NoteAuthenticationInfo::note_ids()`).